### ISSUE TITLE
enhancement: make it easier to create SDATA from value pairs

### ISSUE THREAD

**Issue by czanik** [balabit/syslog-ng#974]
Right now it's difficult to create custom SDATA fields for the syslog() destination. Each needs to be created one by one in the configuration file with a rewrite/set rule: https://www.balabit.com/sites/default/files/documents/syslog-ng-ose-3.7-guides/en/syslog-ng-ose-v3.7-guide-admin/html-single/index.html#custom-sdata-fields

I'd propose to add an add-sdata() option to the syslog() destination with a syntax like the JSON template function has for nv pair selection. Of course, not all "--scope" parameters make sense here, as syslog related fields are already included (it's a syslog message anyway...).

Use case: when using patterndb or especially the key value parser, it can take a lot of effort (or simply impossible) to know all of the possible name value pairs in advance. If we have a client - relay - server architecture where parsing is done on the relays, make it easier (or possible at all) to forward all the name value pairs created by the parsers to the central server.

**Comment by faxm0dem** [comment]
I'm all for simplifying this. I'd go even further in the direction already discussed on the list a few months ago: it should be easy to have **all** fields transfered from one syslog-ng host to another using IETF, even `TAGS`.

**Comment by bazsi** [comment]
I agree with both points. SDATA right now is really a bit outside of the
normal syslog flow, and they should be more integrated.

I agree with @czanik that an sdata() option that supports value-pairs
should be added.

And I also agree with @faxm0dem. We should be able to define a
mapping/schema of our internal name-value pairs to a transport mechanism
(which can be RFC5424 but not necessarily just that) to make it easy to
transfer all parsed state to a next-hop syslog server in one easy step.

I think these are two separate issues though.

-- 
Bazsi
